code/drasil-theory/lib/Theory/Drasil/MultiDefn.hs

```diff
@@ -1,59 +1,95 @@
-{-# LANGUAGE TemplateHaskell, Rank2Types, ScopedTypeVariables, PostfixOperators  #-}
+{-# LANGUAGE PostfixOperators #-}
+{-# LANGUAGE Rank2Types #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TupleSections #-}
 
 -- | Defines types and functions for creating mult-definitions.
-module Theory.Drasil.MultiDefn (
-  -- * Types
-  MultiDefn, DefiningExpr,
-  -- * Constructors
-  mkMultiDefn, mkMultiDefnForQuant, mkDefiningExpr,
-  -- * Functions
-  multiDefnGenQD, multiDefnGenQDByUID) where
-
-import Control.Lens ((^.), view, makeLenses)
+module Theory.Drasil.MultiDefn
+  ( -- * Types
+    MultiDefn,
+    DefiningExpr,
+
+    -- * Constructors
+    mkMultiDefn,
+    mkMultiDefnForQuant,
+    mkDefiningExpr,
+
+    -- * Functions
+    multiDefnGenQD,
+    multiDefnGenQDByUID,
+  )
+where
+
+import Control.Lens (makeLenses, view, (^.))
 import Data.List (union)
 import qualified Data.List.NonEmpty as NE
-
 import Language.Drasil hiding (DefiningExpr)
 import Language.Drasil.Development (showUID)
 
 -- | 'DefiningExpr' are the data that make up a (quantity) definition, namely
 --   the description, the defining (rhs) expression and the context domain(s).
 --   These are meant to be 'alternate' but equivalent definitions for a single concept.
-data DefiningExpr e = DefiningExpr {
-  _deUid  :: UID,            -- ^ UID
-  _cd     :: [UID],          -- ^ Concept domain
-  _rvDesc :: Sentence,       -- ^ Defining description/statement
-  _expr   :: e  -- ^ Defining expression
-}
+data DefiningExpr e = DefiningExpr
+  { -- | UID
+    _deUid :: UID,
+    -- | Concept domain
+    _cd :: [UID],
+    -- | Defining description/statement
+    _rvDesc :: Sentence,
+    -- | Defining expression
+    _expr :: e
+  }
+
 makeLenses ''DefiningExpr
 
-instance Eq            (DefiningExpr e) where a == b = a ^. uid == b ^. uid
-instance HasUID        (DefiningExpr e) where uid    = deUid
-instance ConceptDomain (DefiningExpr e) where cdom   = (^. cd)
-instance Definition    (DefiningExpr e) where defn   = rvDesc
-
--- | 'MultiDefn's are QDefinition factories, used for showing one or more ways we
---   can define a QDefinition.
-data MultiDefn e = MultiDefn {
-    _rUid  :: UID,                                      -- ^ UID
-    _qd    :: QuantityDict,                             -- ^ Underlying quantity it defines
-    _rDesc :: Sentence,                                 -- ^ Defining description/statement
-    _rvs   :: NE.NonEmpty (DefiningExpr e) -- ^ All possible/omitted ways we can define the related quantity
-           -- TODO: Why is this above constraint redundant according to the smart constructors?
-}
+instance Eq (DefiningExpr e) where a == b = a ^. uid == b ^. uid
```

Comment:
Same here, the previous layout of the `instance` and `data MultDefn` were nicer.

Reply:
Fixed.